Source/Core/DolphinQt/Achievements/AchievementHeaderWidget.cpp

```diff
@@ -80,6 +84,24 @@ void AchievementHeaderWidget::UpdateData()
     return;
   }
 
+  const AchievementManager::BadgeStatus& player_badge =
+      AchievementManager::GetInstance()->GetPlayerBadge();
+  m_user_icon->setVisible(false);
+  m_user_icon->clear();
+  m_user_icon->setText({});
+  if (Config::Get(Config::RA_BADGES_ENABLED) && player_badge.loaded)
```

Comment:
This is minor but if the badge isn't loaded yet and the user has the AchievementHeaderWidget` open, how will they see the icon updated?  Do they have to manually close and refresh the screen?

Reply:
Every badge fetch is followed by a call to m_update_callback(), which is currently set to UpdateData in the parent object which then calls updates on all the widgets inside it.